proposals/dictionary-expressions.md

```diff
@@ -220,33 +281,48 @@ The new rules above represent a breaking change: For types that are a valid conv
+
+Pointing at the existing method would have the positive that it would add to the BCL surface area.  But it would have the negative of forcing going through interfaces to create an immutable dictionary.
 
-It is common for dictionaries to take in a `comparer` value, to determine how keys are compared when adding, removing, and looking them up.  Use in the ecosystem is prevalent, and much discussion and feedback from the community is that being able to supply a comparer is important to them.  How could we accomplish this with the new dictionary-expression?  Options include:
+Pointing at a new method would allow for more efficient, stack-based, initialization.  However, it would also increase the BCL surface area.
 
-1. Provide no support (or no support in C# 13), leaving such dictionaries as ones you could not use *dictionary expressions* for.
+BCL might also prefer something like `CreateRange(ReadOnlySpan<TKey> keys, ReadOnlySpan<TValues> values)` such a signature would then allow constant data segments to be pointed at with an dictionary expression like so:
 
-1. Provide a special syntactic form *solely* for the purpose of supplying that value.  For example: `[comparer: myComp, "mads": 21, .. ldmMembers]`.  Here `comparer` would be a contextual keyword.  Users wanting to use that as an actual key would say `@comparer: 1`
+```c#
+FrozenDictionary<int, int> mapping = [1: 2, 2: 4, 4: 8];
 
-1. Provide a special syntactic form for the purpose of supplying data to constructors and create methods.  For example: `[new: (comparer: myComp, capacity: 50), "mads": 21, .. ldmMembers]`
+// maps to:
+// No stack allocs at all!
+private static ReadOnlySpan<int> __keys => [1, 2, 4];
+private static ReadOnlySpan<int> __values => [2, 4, 8];
+mapping = FrozenDictionary.CreateRange(__keys, __values);
+```
 
 ## Construction
 
 > The elements of a collection expression are evaluated in order, left to right. Each element is evaluated exactly once, and any further references to the elements refer to the results of this initial evaluation.
-> 
+>
+> If the collection expression includes a `comparer: expression`, that `expression` will be evaluated before the subsequence elements are evaluated, and will be converted to an `IEqualityComparer<TKey>`.  If present:
+>
+> 1. If using a constructor to instantiate the value, the constructor must take an `IEqualityComparer<TKey>` as the sole parameter which the converted expression will be passed to.
+> 2. If using a `create method`, the method must take an `IEqualityComparer<TKey>` as the first parameter which the converted parameter will be passed to.
+> 3. If creating an interface, the converted expression will be used as the equality comparer controlling the behavior of the final type (synthesized or otherwise) instantiated.
+>
 > ```diff
```

Comment:
Let's use bold for additions rather than `diff`:
> **A key_value_pair_element evaluates its interior expressions in order, left to right.  In other words, the key is evaluated before the value.**
>
> For each element in order:
>
> - If the element is an expression element, the applicable Add instance or extension method is invoked with the element expression as the argument. (Unlike classic collection initializer behavior, element evaluation and Add calls are not necessarily interleaved.)
> 
>     **If the target is a dictionary type, then the element must be a `KeyValuePair<,>`.  The applicable indexer is invoked with the `.Key` and `.Value` members of that pair.**
>
> - If the element is a spread element then one of the following is used:
>   - An applicable GetEnumerator instance or extension method is invoked on the spread element expression and for each item from the enumerator the applicable Add instance or extension method is invoked on the collection instance with the item as the argument. If the enumerator implements IDisposable, then Dispose will be called after enumeration, regardless of exceptions.
> 
>     **If the target is a dictionary-type, the enumerator's element type must be some `KeyValuePair<,>`, and for each of those elements the applicable indexer is invoked on the collection instance with the `.Key` and `.Value` members of that pair.**